monitoredresources/monitored_resources.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitoredresources
+
+import (
+	"fmt"
+	"os"
+	"sync"
+	)
+
+type MonitoredResInterface interface {
+	GetType() string
+	GetLabels() map[string]string
+}
+
+type GKEContainer struct {
+	ProjectID     string
+	InstanceID    string
+	ClusterName   string
+	ContainerName string
+	NamespaceID   string
+	PodId         string
+	Zone          string
+}
+
+func (gke *GKEContainer) GetType() string {
+	return "gke_container"
+}
+
+func (gke *GKEContainer) GetLabels() map[string]string {
+	labels := map[string]string{
+		GKELabelProjectID:     gke.ProjectID,
+		GKELabelInstanceID:    gke.InstanceID,
+		GKELabelZone:          gke.Zone,
+		GKELabelClusterName:   gke.ClusterName,
+		GKELabelContainerName: gke.ContainerName,
+		GKELabelNamespaceID:   gke.NamespaceID,
+		GKELabelPodID:         gke.PodId,
+	}
+	return labels
+}
+
+type GCEInstance struct {
+	ProjectID  string
+	InstanceID string
+	Zone       string
+}
+
+func (gce *GCEInstance) GetType() string {
+	return "gce_instance"
+}
+
+func (gce *GCEInstance) GetLabels() map[string]string {
+	labels := map[string]string{
+		GCELabelProjectID:  gce.ProjectID,
+		GCELabelInstanceID: gce.InstanceID,
+		GCELabelZone:       gce.Zone,
+	}
+	return labels
+}
+
+type AWSEC2Instance struct {
+	AWSAccount string
+	InstanceID string
+	Region     string
+}
+
+func (aws *AWSEC2Instance) GetType() string {
+	return "aws_ec2_instance"
+}
+
+func (aws *AWSEC2Instance) GetLabels() map[string]string {
+	labels := map[string]string{
+		AWSEC2LabelAwsAccount: aws.AWSAccount,
+		AWSEC2LabelInstanceID: aws.InstanceID,
+		AWSEC2LabelRegion:     aws.Region,
+	}
+	return labels
+}
+
+var mr MonitoredResInterface
```

Comment:
`mr` is a bit unclear, maybe call this something like `autoDetected`?

Reply:
changed it to autoDetected.